src/Control/Lens/Fold.hs

```diff
@@ -279,6 +280,14 @@ replicated n0 f a = go n0 where
   go n = m *> go (n - 1)
 {-# INLINE replicated #-}
 
+-- | Use a 'Fold' to work over part of a structure. This is a version of 'aside' that takes a Fold
+-- instead. The resulting 'Fold' has as many targets as the provided 'Fold'.
+--
+-- >>> [Just ('a', 10), Nothing, Just ('b', 20)] ^.. folded . beneath (_Just . _1)
+-- [(Just ('a',10),'a'),(Just ('b',20),'b')]
+beneath :: Fold s a -> Fold s (s, a);
```

Comment:
extra `;`.

The first argument should be first-order. (E.g. `foldrOf :: Getting (Endo r) s a -> (a -> r -> r) -> r -> s -> r`)

I don't understand how this is related to `aside`. Then the type should be

```haskell
beneath :: Fold s a -> Fold (e, s) (e, a)
```

Reply:
What do you mean by `the first argument should be first-order`?

You are right, the change of type with regards to `aside` was a bit accidental, I twisted it a bit further because it fit my usage pattern. Perhaps this is up-to-debate, since plumbing with some `to dup` (see example below) is possible too.

The main motivation is a sort of filtering a'la mapMaybe, say you have some
```haskell
let xs = [ Just (Right 'a', 10)
         , Just (Left 'b', 20)
         , Nothing
         , Just (Left 'c', 30)
         ]
```

And you want to fold over all the things that happen to have a Left deep inside, together with the matching value. 

Then you could write say
```haskell
matching = xs ^.. folded . beneath (_Just . _1 . _Left)
-- or if we assumed `beneath' :: Fold s a -> Fold (e, s) (e, a)`
matching' = xs ^.. folded . to dup . beneath' (_Just . _1 . _Left)

```

With `aside` we'd require repetitive labor:

```haskell
dup x = (x, x)
matching = xs ^.. folded . to dup . aside _Just . alongside id _1 . aside _Left
```


The path into the "sub-focus" (the Fold you'd pass to beneath) would require interleaving aside and alongside in appropriate numbers until you get to the focus like in the example above.